strawberryfields/program.py

```diff
@@ -572,7 +572,15 @@ def _get_compiler(compiler_or_name):
         if device is not None and device.gate_parameters:
             bb_device = bb.loads(device.layout)
             bb_compiled = sf.io.to_blackbird(compiled)
-            user_parameters = match_template(bb_device, bb_compiled)
+
+            try:
+                user_parameters = match_template(bb_device, bb_compiled)
+            except bb.utils.TemplateError:
+                raise CircuitError(
+                    "Program cannot be used with the compiler '{}' "
+                    "due to incompatible topology.".format(compiler.short_name)
```

Comment:
Just for my own understanding: would it be sure that the reason is always `incompatible topology`? ([Seems like there could be other cases for `TemplateError` too](https://github.com/XanaduAI/blackbird/blob/60e1ae43cd97e8759496a5ecc177b16f9b0b8299/blackbird_python/blackbird/utils.py#L187), though I'd imagine that these are unlikely to arise here? :thinking: )

Reply:
Oh, good catch @antalszava. Although, I would say from looking at the source code that the first three are unlikely to occur (possible even impossible to occur), since, by construction,

* Both arguments will always be a template
* This function uses the same version of BB to compute both arguments